classes/ReassuranceActivity.php

```diff
@@ -0,0 +1,175 @@
+            'date_upd' => array('type' => self::TYPE_DATE, 'shop' => true, 'validate' => 'isDate'),
+        )
+    );
+
+    /**
+     * getAllBlockByLang
+     *
+     * @param  int $id_lang
+     * @param  int $id_shop
+     *
+     * @return array
+     */
+    public static function getAllBlockByLang($id_lang = 1, $id_shop = 1)
+    {
+        $sql = 'SELECT * FROM `'._DB_PREFIX_.'psreassurance` pr
+            LEFT JOIN '._DB_PREFIX_.'psreassurance_lang prl ON (pr.id_psreassurance = prl.id_psreassurance)
+            WHERE prl.id_lang = "'.(int)$id_lang.'" AND prl.id_shop = "'.(int)$id_shop.'"
+            ORDER BY pr.position';
+
+        $result = Db::getInstance()->executeS($sql);
+
+        return $result;
+    }
+
+    public function handleBlockValues($psr_languages, $type_link)
+    {
+        $languages = Language::getLanguages();
+        $newValues = array();
+
+        foreach ($psr_languages as $key => $value) {
+            $newValues[$key] = array();
+            $newValues[$key]['title'] = $value->title;
+            $newValues[$key]['description'] = $value->description;
+            $newValues[$key]['url'] = $value->url;
+        }
+
+        foreach ($languages as $language) {
+            if (false === array_key_exists($language['id_lang'], $newValues)) {
+                continue;
+            }
+            $this->title[$language['id_lang']] = $newValues[$language['id_lang']]['title'];
+            $this->description[$language['id_lang']] =  $newValues[$language['id_lang']]['description'];
+            if ($type_link == 2) {
+                $this->link[$language['id_lang']] =  $newValues[$language['id_lang']]['url'];
+            } else {
+                $this->link[$language['id_lang']] = '';
+            }
+        }
+
+        if (isset($id_cms) && $type_link == 1) {
+            $this->id_cms = $id_cms;
+            $link = Context::getContext()->link;
+            foreach ($languages as $language) {
+                $this->link[$language['id_lang']] = $link->getCMSLink($id_cms, null, null, $language['id_lang']);
+            }
+        }
+
+        if ($type_link == 'undefined') {
+            $type_link = 0;
+        }
+        $this->type_link = $type_link;
+    }
+
+    /**
+     * getAllBlockByShop
+     *
+     * @param  int $id_shop
+     *
+     * @return array
+     */
+    public static function getAllBlockByShop($id_shop = 1)
+    {
+        $return = array();
+        $sql = 'SELECT * FROM `'._DB_PREFIX_.'psreassurance` pr
+            LEFT JOIN '._DB_PREFIX_.'psreassurance_lang prl ON (pr.id_psreassurance = prl.id_psreassurance)
+            WHERE prl.id_shop = "'.(int)$id_shop.'"
+            GROUP BY prl.id_lang, pr.id_psreassurance
+            ORDER BY pr.position';
+
+        $result = Db::getInstance()->executeS($sql);
+
+        foreach ($result as $key => $value) {
+            $return[$value['id_lang']][$value['id_psreassurance']]['title'] = $value['title'];
```

Comment:
Doesn't this raise a PHP warning ?
I think you're missing a `$return[$value['id_lang']] = array();` here

Reply:
0 notice 0 warning

Reply:
agreed with @matks it's cleaner to init your array

Reply:
it raise an error en php < 7. yes array must be initialized